bugalink-backend/trips/views.py

```diff
@@ -81,3 +88,46 @@ def reject(self, request, *args, **kwargs):
         trip_request.status = "REJECTED"
         trip_request.save()
         return Response(self.get_serializer(trip_request).data)
+
+
+class TripRecommendationViewSet(
+    mixins.RetrieveModelMixin,
+    mixins.DestroyModelMixin,
+    viewsets.GenericViewSet,
+):
+    queryset = Trip.objects.all()
+    serializer_class = TripSerializer
+
+    @action(detail=True, methods=["get"])
+    def get(self, request, user_id, *args, **kwargs):
+        try:
+            data = {"trips": []}
+            for passenger_routine in PassengerRoutine.objects.filter(
+                passenger__user_id=user_id
+            ):
+                data["trips"] = (
+                    data["trips"]
+                    + PassengerRoutineSerializer(passenger_routine)
+                    .get_recommendations(obj=passenger_routine)
```

Comment:
A serializer's purpose is to serialize the objects as JSON as the name suggests, they should not have a method for computing recommendations. This doesn't comply with the single-responsability principle in software design